src/Game.java

```diff
@@ -1,45 +1,119 @@
+      numPlayersString = scanner.nextLine();
+
+      if (numPlayersString.length() == 0){
+        System.out.println("Invalid input, please insert a number:");
+      } else {
+        numPlayers = Integer.parseInt(numPlayersString);
+        if (numPlayers > 1) {
+          break;
+        } else if (numPlayers == 1) {
+          System.out.println(
+              "You cannot play alone! Please enter a number greater than 1:");
+        } else {
+          System.out.println("Please enter a number greater than 1:");
+        }
+      }
+    }
+
+    List<Player> allPlayers = new ArrayList<>();
+
+    for (int i = 0; i < numPlayers; i++) {
+      System.out.println("Please enter the name of player " + (i + 1) + ":");
+      String playerName;
+      while (true) {
+        playerName = scanner.nextLine();
+        if (!allPlayers.stream().map(Object::toString)
+            .collect(Collectors.toList()).contains(playerName)) {
+          if (playerName.length() != 0) {
+            break;
+          }
+          System.out.println("Invalid name, please enter another name");
+        } else {
+          System.out.println("Please enter another name, " + playerName
+              + " has already been taken:");
         }
+      }
 
-        int currentLeaderIndex = 0;
+      allPlayers.add(new Player(playerName));
+    }
+
+    int currentLeaderIndex = 0;
+
+    mainLoop:
+    while (true) {
 
-        while (true) {
-            System.out.println("Do you wish to start a new game? Y/N");
-            String answer = scanner.nextLine();
+      Player leader = allPlayers.get(currentLeaderIndex);
 
-            if (answer.startsWith("N")) {
-                break;
-            }
+      Word word;
+      System.out.println(leader + ", you are the leader, please pick a word:");
 
-            Player leader = allPlayers.get(currentLeaderIndex);
+      while (true) {
+        word = new Word(scanner.nextLine());
+        if (word.wordLength() != 0) {
+          break;
+        }
+        System.out.println("Please enter a word with at least 1 letter:");
+      }
+
+      List<Player> players = new ArrayList<>(allPlayers);
+      players.remove(leader);
 
-            System.out.println(leader + ", you are the leader, please pick a word:");
+      Session session = new Session(scanner, players, leader, word);
+      session.play();
 
-            Word word = new Word(scanner.nextLine());
+      currentLeaderIndex = (currentLeaderIndex + 1) % allPlayers.size();
+      System.out.println("Current Standings: ");
+      System.out.println(leaderboardToSet(allPlayers));
 
-            List<Player> players = new ArrayList<>(allPlayers);
-            players.remove(leader);
+      System.out.println("Do you wish to start a new game? Y/N");
+      String answer = scanner.nextLine();
 
-            Session session = new Session(scanner, players, leader, word);
-            session.play();
+      while (!answer.startsWith("y") && !answer.startsWith("Y")) {
```

Comment:
`askUntilCondition("Do you wish to start a new game? Y/N", input -> input.startsWith("y") || input.startsWith("n"))`

Or you can make a new function, `askYesOrNo(String message, Runnable yesAction, Runnable noAction)` for this type of thing.